coordinatorlib/src/main/scala/com/socrata/datacoordinator/truth/json/JsonColumnRep.scala

```diff
@@ -13,6 +13,16 @@ trait JsonColumnReadRep[CT, CV] extends JsonColumnCommonRep[CT, CV] {
 
 trait JsonColumnWriteRep[CT, CV] extends JsonColumnCommonRep[CT, CV] {
   def toJValue(value: CV): JValue
+
+  /**
+    * Output of upsert mutation are in the form:
+    *   [{"typ":"insert","id": $JVALUE,"ver":"rv-vhb7.vn49.4xy9"}]
+    * Libraries were written to only handle "id" as json string.
+    * This allows complex type like URL to have a different/simplified json format when it is used
+    * as return id value in upsert mutation.
+    */
+  def toRowIdJValue(value: CV): JValue = toJValue(value)
```

Comment:
The point of returning the row IDs here is so that user code that cares can align their idea of the value with the version that also come back (or the error, if the row update failed).  Returning only part of the value seems to defeat that purpose.

Reply:
Note BTW that no user code cares about this value, because it's eaten by core.  If we change it so that it's not guaranteed to be a string, we don't have to worry about breaking anyone other than ourselves.

Reply:
changed.  Will require change in soda-java client/core reference.  Wouldn't it affect people who use soda-java client?  This use case is small enough that I guess breaking/requiring update is fine.

Reply:
As far as I'm aware, there is no way for people who use the soda-java client to get this response.